### DISCUSSION TITLE
Database to store Chats

### DISCUSSION THREAD

**Discussion by abhi-kyros** [aliyazdi75/flutter_chat#7] [General]
Which database should be use to store all chats. Your repo is one of the best and cleanest ever found thank you so much.

**Comment by aliyazdi75** [comment]
Hi, thanks for your compliment. I'm so glad that you found it useful.

In my opinion, using a simple [Sqflite](https://pub.dev/packages/sqflite) would be a good choice, since you need tables to structure your chats and use them when you are offline.

Share your thoughts if you have some other options.

**Reply by abhi-kyros** [reply]
I had used hive database and it is working fine. I just have one question more that when app is in background for so long it doesn't reconnect so message are unable to get. I am facing this issue, have you faced something like that? if yes what steps have you took to maintain that state? Thank you.

**Reply by aliyazdi75** [reply]
Yes, the hive is good too. For background, there are two states, the first one(when you hit the middle button) your application works fine and doesn't need to do more. But if you hit back bottom, the app will be closed and you have no access to your app. You should control your new message with firebase services like notification. You have to add firebase to your app and when a new message arrives, shows the new notification.

**Reply by abhi-kyros** [reply]
I am facing one issue, sometimes _hubConnection is getting null when sense offline to online or in background for long time. other than that it is working fine.  Do you have any solution regarding that?
